### PR TITLE
Feature/UI components

### PR DESCRIPTION
**Pull request by ZZH0C**

### CODE REVIEW

File: src/components/Head/Head.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { HeaderLogo } from '../Header_logo/HeaderLogo';
+import { HeaderUser } from '../Header_user/HeaderUser';
+
+type HeadProps = React.HTMLAttributes<HTMLElement>;
+
+export const Head: React.FC<HeadProps> = () => {
```
Comment (bondarchukse) [inline src/components/Head/Head.tsx:7]: Use `children` property to make your component more flexible. You can pass `HeaderUser` as children! As for the logo would be great to have logo url property for rendering logo at the header (to render logo if url set). 
The most common action for the logo is navigation to the home page, so make sence to wrap this element with router link

File: src/components/Header_logo/HeaderLogo.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { HeaderLogo } from './HeaderLogo';
+import { ComponentMeta, ComponentStory } from '@storybook/react';
+import React from 'react';
+
+export default {
```
Comment (bondarchukse) [inline src/components/Header_logo/HeaderLogo.stories.tsx:5]: All files and directories names must follow the naming convention, so `Header_logo` name is incorrect!
Only one (sometimes two) naming styles using for a single project 
We should use `PascalCase` for the components (and for the component folders) and `camelCase` for other staff (like helpers, utils and ect)

File: src/components/Header_logo/HeaderLogo.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+type HeaderLogoProps = React.HTMLAttributes<HTMLElement>;
+
+export const HeaderLogo: React.FC<HeaderLogoProps> = () => {
+  return (
+    <div className="logoBlock">
+      <a href="index.html">
+        <img alt="1" src="images/logotype.png" className="logo" title="Logo" />
```
Comment (bondarchukse) [inline src/components/Header_logo/HeaderLogo.tsx:9]: `alt`, `src` and `title` attributes should not be hardcoded there!
`Page Title` text should be a children content

File: src/components/Header_user/HeaderUser.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { HeaderUserIcons } from '../Header_user_icons/HeaderUserIcons';
+type HeaderUserProps = React.HTMLAttributes<HTMLElement>;
+
+export const HeaderUser: React.FC<HeaderUserProps> = () => {
```
Comment (bondarchukse) [inline src/components/Header_user/HeaderUser.tsx:5]: component should not contains hardcoded data!

File: src/components/Header_user_icons/HeaderUserIcons.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+// import '../../styles/main.css';
+
+interface HeaderUserIconsProps {
+  style: 'headerIcon icon-mail' | 'headerIcon icon-bell';
```
Comment (bondarchukse) [inline src/components/Header_user_icons/HeaderUserIcons.tsx:5]: try to use more common variables for the icon. 
1. `style` - is a default property for the react component. Your property name is confusing for other developers
2. You should extend props from the base react props to able pass custom className and other staff
3. You should use `classnames` library for class names concatination

interface HeaderUserIconsProps extend SomeCommonHtmlTagProps {
 
 iconName: 'icon-mail' | 'icon-bell';

 url?: string;

}

````
export const HeaderUserIcons: React.FC<HeaderUserIconsProps> = ({ iconName, url, className, ...props }) => {

const icon = useMemo(() => (
  <i className={classNames('headerIcon', iconName)} />
), [iconName])

return url
 ? (<a href={url} className="classNames('btn', 'btn-sm', 'btn-header', className)" {...props}>{icon}</a>)
 : (<span className="classNames('btn', 'btn-sm', 'btn-header', className)" {...props}>{icon}</span>);
};
````

File: src/components/Mainsection/MainSection.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { MainSectionSubmenu } from './Mainsection_submenu/MainSectionSubmenu';
+import { MainSectionContent } from './Mainsection_content/MainSectionContent';
+type MainSectionProps = React.HTMLAttributes<HTMLElement>;
+
+export const MainSection: React.FC<MainSectionProps> = () => {
```
Comment (bondarchukse) [inline src/components/Mainsection/MainSection.tsx:6]: Use `children` here!!!

File: src/components/Mainsection/Mainsection_content/MainSectionContent.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { MainSectionContentHeading } from './Mainsection_content_heading/MainSectionContentHeading';
+import { MainSectionContentSearchbar } from './Mainsection_content_searchbar/MainSectionContentSearchBar';
+import { MainSectionContentPosts } from './Mainsection_content_items_container/MainSectionContentPosts';
+type MainSectionContentProps = React.HTMLAttributes<HTMLElement>;
+
+export const MainSectionContent: React.FC<MainSectionContentProps> = () => {
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_content/MainSectionContent.tsx:7]: Use `children` here!!!

File: src/components/Mainsection/Mainsection_content/Mainsection_content_heading/MainSectionContentHeading.tsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+type MainSectionContentHeadingProps = React.HTMLAttributes<HTMLElement>;
+
+export const MainSectionContentHeading: React.FC<MainSectionContentHeadingProps> =
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_content/Mainsection_content_heading/MainSectionContentHeading.tsx:4]: Use `children` here!!!

File: src/components/Mainsection/Mainsection_content/Mainsection_content_items_container/MainSectionContentPosts.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { MainSectionContentItem } from './Mainsection_content_post/MainSectionContentItem';
+import { MainSectionContentFooter } from './Mainsection_content_footer/MainSectionContentFooter';
+type MainSectionContentPostsProps = React.HTMLAttributes<HTMLElement>;
+
+export const MainSectionContentPosts: React.FC<MainSectionContentPostsProps> =
+  () => {
+    const examplePosts = [
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_content/Mainsection_content_items_container/MainSectionContentPosts.tsx:8]: components should not contains hardcode or mock data!!!

File: src/components/Mainsection/Mainsection_content/Mainsection_content_items_container/Mainsection_content_footer/MainSectionContentFooter.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+type MainSectionContentFooterProps = React.HTMLAttributes<HTMLElement>;
+
+export const MainSectionContentFooter: React.FC<MainSectionContentFooterProps> =
+  () => {
+    return (
+      <footer className="footer">
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_content/Mainsection_content_items_container/Mainsection_content_footer/MainSectionContentFooter.tsx:7]: needs to make decomposition of that code! No hardcode! 
`<a href="index.html" title="menu">` is not good solution for the SPA application. It should be router link!

File: src/components/Mainsection/Mainsection_content/Mainsection_content_items_container/Mainsection_content_post/MainSectionContentItem.tsx [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+) => {
+  const tags = props.tags.map((elem) => {
+    return (
+      <a key={Math.random()} href="index.html" className="tag" title="tag">
+        {elem}
+      </a>
+    );
+  });
+
+  const d = new Date(props.dateTime);
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_content/Mainsection_content_items_container/Mainsection_content_post/MainSectionContentItem.tsx:25]: you shouldn't minify variable names!

File: src/components/Mainsection/Mainsection_submenu/MainSectionSubmenu.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    { name: 'subitem 3', href: 'index.html', active: false },
+  ];
+  const testPropsActive = [
+    { name: 'subitem 1', href: 'index.html', active: false },
+    { name: 'subitem 2', href: 'index.html', active: true },
+    { name: 'subitem 3', href: 'index.html', active: false },
+  ];
+  return (
+    <aside className="submenu">
+      <MainSectionSubmenuItem name={'Item 1'} content={testProps} />
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_submenu/MainSectionSubmenu.tsx:18]: `children` should be there!

File: src/components/Mainsection/Mainsection_submenu/MainSectionSubmenu.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { MainSectionSubmenuItem } from './Mainsection_submenu_item/MainSectionSubmenuItem';
+type MainSectionSubmenuProps = React.HTMLAttributes<HTMLElement>;
+
+export const MainSectionSubmenu: React.FC<MainSectionSubmenuProps> = () => {
+  const testProps = [
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_submenu/MainSectionSubmenu.tsx:6]: no mocks inside components!

File: src/components/Mainsection/Mainsection_submenu/MainSectionSubmenu.tsx [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+  ];
+  const testPropsActive = [
+    { name: 'subitem 1', href: 'index.html', active: false },
+    { name: 'subitem 2', href: 'index.html', active: true },
+    { name: 'subitem 3', href: 'index.html', active: false },
+  ];
+  return (
+    <aside className="submenu">
+      <MainSectionSubmenuItem name={'Item 1'} content={testProps} />
+      <MainSectionSubmenuItem name={'Item 2'} content={testPropsActive} />
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_submenu/MainSectionSubmenu.tsx:19]: no hardcode!

File: src/components/Mainsection/Mainsection_submenu/Mainsection_submenu_item/MainSectionSubmenuItem.tsx [hunk 15]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+interface MainSectionSubmenuItemProps
+  extends React.HTMLAttributes<HTMLElement> {
+  name: string;
+  content: { name: string; href: string; active: boolean | undefined }[];
```
Comment (bondarchukse) [inline src/components/Mainsection/Mainsection_submenu/Mainsection_submenu_item/MainSectionSubmenuItem.tsx:6]: this is a bad solution for rendering menu items. You will never store menu structure as an array here. Children would be much better

File: src/components/Navbar/Navbar_elem/Navbarelem.tsx [hunk 16]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+interface NavbarElemProps extends React.HTMLAttributes<HTMLElement> {
+  isActive: true | false;
+  text: string;
+}
+
+export const NavbarElem: React.FC<NavbarElemProps> = ({ isActive, text }) => {
+  let style;
```
Comment (bondarchukse) [inline src/components/Navbar/Navbar_elem/Navbarelem.tsx:9]: use `classnames` instead

File: src/components/Navbar/Navbar_elem/Navbarelem.tsx [hunk 17]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+interface NavbarElemProps extends React.HTMLAttributes<HTMLElement> {
+  isActive: true | false;
```
Comment (bondarchukse) [inline src/components/Navbar/Navbar_elem/Navbarelem.tsx:4]: true | false === boolean

**Review by Allakazam — APPROVED** [review]

**Pull request merged**